docs: Added contributing guide (#29)
jdwilkin4/fcc-forum-clone: CONTRIBUTING.md

### Picking up an issue to work on

Browse through [opened issues](https://github.com/jdwilkin4/fcc-forum-clone/issues). Feel free to start working on any issue which does not have have assignee yet!

If you are [collaborator](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-personal-account-settings/permission-levels-for-a-personal-account-repository#collaborator-access-for-a-repository-owned-by-a-personal-account) - assign yourself to the issue you want to work on.

If you are not collaborator - would be better to left a comment in the issue thread, so we know that you are working on it.

In any case - make sure to use the issue thread to ask questions about the issue if you need!

## Making changes

If you are experienced with Git workflow, **How to contribute** section of our [README](./README.md) may be enough to work on the issue you picked up!

If you are somewhat new to Git and GitHub - in the below will be couple more tips and useful links about all this!

### Installing git

If you don't have git on your machine, [install it](https://help.github.com/articles/set-up-git/).

### Installing dependencies

> Instructions about this can be found in [README](./README.md).

Reviewer: This section should be moved further down. It should be after the section about cloning the project. 
Also let's give them some instructions on how to install Node.js in order to use the `npm` command. 

Proposed fix: 
If you don't have Node.js installed on your computer, please read through [these instructions](https://nodejs.dev/en/).  If you are not sure if Node.js is installed, please run the `node -v` command in the command line. 

Then change to the project directory by using the `cd fcc-forum-clone` command. To install the dependencies, run the `npm install` command.

### Forking the repo

Fork this repository by clicking on the fork button on the top of this page.
This will create a copy of this repository in your account.

- Using GitHub Desktop:
  - [Getting started with GitHub Desktop](https://docs.github.com/en/desktop/installing-and-configuring-github-desktop/getting-started-with-github-desktop) will guide you through setting up Desktop.

>   - Once Desktop is set up, you can use it to [fork the repo](https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/cloning-and-forking-repositories-from-github-desktop)!

Reviewer: I don't think we need the exclamation point here since this sentence doesn't seem to indicate strong feelings or strong emphasis.

> ### Creating local copy of the repo (aka cloning)

Reviewer: ```suggestion
### Creating a local copy of the repo (aka cloning)
```

Now clone the forked repository to your machine. Go to your GitHub account, open the forked repository, click on the code button and then click the copy to clipboard icon.

Open a terminal and run the following git command:

```
git clone url-you-just-copied
```

where `url-you-just-copied` is the url to this repository (your fork of this project).

> ### Creating new branch to start dealing with the issue

Reviewer: ```suggestion
### Creating a new branch to start working on the issue
```

- Enable the checkbox to [allow maintainer edits](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) so the branch can be updated for a merge.
  Once you submit your PR, a team member will review your proposal. We may ask questions or request for additional information.
- We may ask for changes to be made before a PR can be merged, either using [suggested changes](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/incorporating-feedback-in-your-pull-request) or pull request comments. You can make any other changes in your fork, then commit them to your branch.
- As you update your PR and apply changes, mark each conversation as [resolved](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/commenting-on-a-pull-request#resolving-conversations).
- If you run into any merge issues, checkout this [git tutorial](https://github.com/skills/resolve-merge-conflicts) to help you resolve merge conflicts and other issues.

### Still a little bit lost?

> - Check out [this video](https://www.youtube.com/watch?v=RGOj5yH7evk) from freeCodeCamp youTube channel Git and GitHub basics.

Reviewer: ```suggestion
- Check out [this video](https://www.youtube.com/watch?v=RGOj5yH7evk) from freeCodeCamp's YouTube channel on Git and GitHub basics.
```

> - Don't be shy to ask us about anything related to the project and working on it! Depending on your question, use pull request threads, issue threads or the discussions!

Reviewer: I don't think we need either exclamation point here. Periods would work best here.

# freeCodeCamp forum clone's contributing guide

## Documentation

Please, make sure to read through the following documents:

- our [code of conduct](./CODE_OF_CONDUCT.md)
- and [README](./README.md)

## Issues

### Creating issues

> Feel free to check out [our codebase](https://github.com/jdwilkin4/fcc-forum-clone). If you think, that something could be added, fixed, or improved on, first that there is no open issue for that by looking through the [issues tab](https://github.com/jdwilkin4/fcc-forum-clone/issues).

Reviewer 2: Rephrase: If you think something could be added, fixed, or improved, firstly check that there is no open issue already. You can find it in the [Issues](https://github.com/jdwilkin4/fcc-forum-clone/issues) section of this repository.
